### ISSUE TITLE
El Dorado OrderDetail.saleprice_changed() should ignore originalprice and rely upon priceauthorizedby

### ISSUE THREAD

**Issue by kbower** [kbower/tickettest1#1674]
_Imported from trac ticket 1674.
Created by **mariam, kbower**
Opened in trac: May 28, 2012, 11:59:40 AM
Last modified in trac: May 28, 2012, 12:13:17 PM_
***
_by mariam, kbower:_
We should redefine the el dorado version of OrderDetail.saleprice_changed() to ignore originalprice and rely upon priceauthorizedby

-------- Original Message --------
Subject:         RE: MD9 Discount
Date:         Mon, 28 May 2012 11:56:55 -0400
From:         Mariam Farach <Mariam@ElDoradoFurniture.com>
To:         Kent Bower <kb@retailarchitects.com>
CC:         dev (dev@retailarchitects.com) <dev@retailarchitects.com>, Jesus Capo <Jesus@ElDoradoFurniture.com>

Kent,

That makes sense.

We can probably go with the "Any time there has been a price override on the line " approach.  That will work for us.

Mariam

*From:*Kent Bower [mailto:kb@retailarchitects.com]
*Sent:* Monday, May 28, 2012 11:10 AM
*To:* Mariam Farach
*Cc:* dev (dev@retailarchitects.com); Jesus Capo
*Subject:* Re: MD9 Discount

It is conflicting with the other rule we implemented for you regarding floor samples.  Site 16 has a promotion record for 133121056 at $229.00.

As you know, if you hover over "Qualifying Products" or "Discount" (Amount), you usually get a clue:

```
SQL> select lineid, productid, packageid,  itemprice + discount as saleprice, ORIGINALPRICE  from orderdetails where orderid = '0527216WX77';

    LINEID PRODUCTID PACKAGEID  SALEPRICE ORIGINALPRICE
---------- --------- --------- ---------- -------------
         1 BAR719905                    0           999
         2 250719904 BAR719905     353.54        353.54
         3 251719905 BAR719905     353.54        353.54
         4 252719906 BAR719905      97.31         97.31
         5 253719907 BAR719905     194.61        194.61
         6 628731362                   99            99
         7 628731362                   99            99
         8 133121056                  349           229
         9 CSA121052                    0          1099
        10 130121053 CSA121052     543.08        543.08
        11 131121054 CSA121052     543.08        543.08
        12 132121055 CSA121052     412.84        412.84

12 rows selected.
```
Look at line 8.  It is no longer being sold at the "original price".  That is to say, the system wanted to sell 133121056 at $229, but since it isn't filled at the store, the special el dorado rule changed the price back to $349, which the system now sees as a price override.  You rewrote the order at site 20 which has no promotion set up.

I suggest maybe for purposes of these discounts, instead of defining a "price override" as any changed price, we define it as any price less than "originalprice" because of the conflict for the floor sample rules.  However, that does leave the loop hole available for a user to override the price all the way down to $229 in this case (for warehouse inventory, for example) and it would still not be seen as a "price override".  We could fix that as well, but only with some deal of effort, I'm hoping it wouldn't be worth it?  Maybe for the long term it would be worth it....

Or else, maybe we define price override more simply as "anytime there has been a price override on the line".  The only drawback there is that if someone overrides a price and then changes it back to what it originally was, the system won't be smart enough to realize there is no longer a price override.

Kent

----
On 5/28/2012 10:39 AM, Mariam Farach wrote:

Good Morning,

Can you please tell me why this order 0527216WX77  is not accepting the MD9 discount?

I created the order again 0528220XH37 and it did take it. The prices before discounts were exactly the same on both the original order, and the reservation I created today.

I have seen this happen before, but the stores had made the adjustments to the price before I got the opportunity to send you the order for your review.

[[Image(None)]]

***
* kbower attached [None](https://trac.retailarchitects.com/trac/attachment/ticket/1674/None) on May 28, 2012, 11:59:40 AM

**Comment by kbower** [comment]
_**kbower** trac comment on May 28, 2012, 12:13:17 PM:_

(Changeset [[2701]](../commit/2ff257b814c1e48442b0c706268f66ad0a8f3e06))
 * closes #1674 (El Dorado OrderDetail.saleprice_changed() should ignore originalprice and rely upon priceauthorizedby)

**Comment by kbower** [comment]
_**kbower** trac status change on May 28, 2012, 12:13:17 PM:_ **closed**
